### ISSUE TITLE
📊Tracking: AWS Chatbot

### ISSUE THREAD

**Issue by SomayaB** [aws/aws-cdk#8636]
Add your +1 👍  to help us prioritize high-level constructs for this service
--- 

### Overview:

AWS Chatbot is an AWS service that enables DevOps and software development teams to use Slack or Amazon Chime chat rooms to monitor and respond to operational events in their AWS Cloud. AWS Chatbot processes AWS service notifications from Amazon Simple Notification Service (Amazon SNS), and forwards them to Slack or Amazon Chime chat rooms so teams can analyze and act on them. Teams can respond to AWS service events from a chat room where the entire team can collaborate, regardless of location. 

[AWS Docs](https://docs.aws.amazon.com/chatbot/index.html)  

### Maturity: CloudFormation Resources Only

See the [AWS Construct Library Module Lifecycle doc](https://github.com/aws/aws-cdk-rfcs/blob/master/text/0107-construct-library-module-lifecycle.md) for more information about maturity levels.

### Implementation:

See the [CDK API Reference](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-chatbot-readme.html) for more implementation details.

### Issue list:

--- 
This is a  📊Tracking Issue

**Comment by asterikx** [comment]
I would love to see:
1. Slack manual approval flows and
2. Slack deployment and health notifiers

There is an existing third-party construct library [@cloudcomponents/cdk-codepipeline-slack](https://github.com/cloudcomponents/cdk-constructs/tree/master/packages/cdk-codepipeline-slack) for this. However, the constructs stopped working as Slack did deprecate some APIs very recently (see my issue [here](https://github.com/cloudcomponents/cdk-constructs/issues/33)).

I feel that build notifications and health alerts through Slack significantly improve developer experience.
It's cumbersome to log in to the console (and deal with incognito tabs for multi-account setups) or check emails all the time to not miss that one important alert.
Slack notifications are seamless and direct.

**Comment by asterikx** [comment]
Higher-level constructs for budget alert notifications would be very useful.

**Comment by humanzz** [comment]
Hello,

In my team, we're CDK users and we've recently started using Chatbot's [SlackChannelConfiguration](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-chatbot-slackchannelconfiguration.html) CloudFormation resource by building up a construct on top of it.

I want to check if you're open to contributions for just this part as the comments above seem to have a potentially larger scope.

**Comment by skinny85** [comment]
> Hello,
> 
> In my team, we're CDK users and we've recently started using Chatbot's [SlackChannelConfiguration](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-chatbot-slackchannelconfiguration.html) CloudFormation resource by building up a construct on top of it.
> 
> I want to check if you're open to contributions for just this part as the comments above seem to have a potentially larger scope.

Absolutely! Take a look at our ['Contributing' guide](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md) for some tips 🙂. Thanks!

**Comment by humanzz** [comment]
@skinny85 just want to point out https://github.com/aws/aws-cdk/issues/9679 which is another issue that was recently created by another willing contributor for the same purpose.
I left its author a comment to ensure we coordinate and avoid duplication of work.

**Comment by skinny85** [comment]
> @skinny85 just want to point out #9679 which is another issue that was recently created by another willing contributor for the same purpose.
> I left its author a comment to ensure we coordinate and avoid duplication of work.

That's great @humanzz , thanks!

**Comment by alexbaileymembr** [comment]
Out of interest, is anyone currently working on an Teams version of the Slack L2 construct (https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-chatbot/lib/slack-channel-configuration.ts)? If not, I'm happy to contribute as we need this currently.

**Comment by airmonitor** [comment]
Hey.

I'm wondering if you could change the priority for AWS ChatBot MS Teams integration.
Currently, the MS team's CFN constructs require an IAM role to be provided, but there is no role if we are creating notifications for CI/CD pipeline using pipelines.CodePipeline.

For Slack, the L2 is doing what is needed to achieve what is needed, but MS Teams is far behind.


Additionally due to recent Slack API changes we are impacted by that, and migrating to MS Teams.

I would be very helpful if L2 for Ms Teams with a corresponding IAM role would be created. That would enable us to use MS Teams notifications for CI/CD pipeline notifications.

**Comment by irobinsonDandH** [comment]
Any update on the MS teams L2 construct for the Chatbot?
